GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/WorkingDirectoryTests.cs

```diff
@@ -451,19 +434,20 @@ public void FolderContentsCorrectAfterCreateNewFolderRenameAndCheckoutCommitWith
             Path.Combine(folder, "MoveUnhydratedFileToDotGitFolder", "Program.cs").ShouldBeAFile(this.fileSystem).WithContents(MoveRenameFileTests.TestFileContents);
         }
 
+        // Case-insensitive filesystems only
         [TestCase, Order(15)]
-        public void FilterNonUTF8FileName()
+        [Category(Categories.CaseInsensitiveFileSystemOnly)]
+        public void FilterNonUTF8FileNameOnCaseInsensitiveFileSystem()
         {
-            string encodingFilename = "ريلٌأكتوبرûمارسأغسطسºٰٰۂْٗ۵ريلٌأك.txt";
-            string folderVirtualPath = this.Enlistment.GetVirtualPathTo("FilenameEncoding");
-
-            this.FolderEnumerationShouldHaveSingleEntry(folderVirtualPath, encodingFilename, null);
-            this.FolderEnumerationShouldHaveSingleEntry(folderVirtualPath, encodingFilename, "ريلٌأكتوبرûمارسأغسطسºٰٰۂْٗ۵ريلٌأك.txt");
-            this.FolderEnumerationShouldHaveSingleEntry(folderVirtualPath, encodingFilename, "ريلٌأكتوبرûمارسأغسطسºٰٰۂْٗ۵ريلٌأك*");
-            this.FolderEnumerationShouldHaveSingleEntry(folderVirtualPath, encodingFilename, "ريلٌأكتوبر*.TXT");
+            this.FilterNonUTF8FileName(false);
+        }
 
-            folderVirtualPath.ShouldBeADirectory(this.fileSystem).WithNoItems("test*");
-            folderVirtualPath.ShouldBeADirectory(this.fileSystem).WithNoItems("ريلٌأكتوب.TXT");
+        // Case-sensitive filesystems only
+        [TestCase, Order(15)]
+        [Category(Categories.CaseSensitiveFileSystemOnly)]
+        public void FilterNonUTF8FileNameOnCaseSensitiveFileSystem()
+        {
+            this.FilterNonUTF8FileName(true);

```

Comment:
Same comment for these new test cases, could we use FileSystemHelpers.CaseSensitiveFileSystem instead of picking one test or another based on `Category`.

Reply:
Also done.  :-)